### ISSUE TITLE
Easy way to ensure all fields are non-null

### ISSUE THREAD

**Issue by Diagoras** [wix/accord#79]
Idiomatic Scala doesn't use null, so it'd be nice for there to be an easy way to guarantee that all the fields of a validated record are non-null without boiler plate `field1 is notNull`, `field2 is notNull`, etc.

**Comment by holograph** [comment]
While you can always [implement a validator](http://wix.github.io/accord/dsl.html#custom-combinators) yourself, from a core library perspective this doesn't seem compelling (I can think of a couple of counter-arguments); implementing this with reflection goes against the grain of Accord, and a typesafe solution is too much work just for this feature.

 Then again, perhaps an "iterator-style" modifier on fields? Then you can do something like `p.fields.each is notNull`, which might be an interesting exercise in using Shapeless (CCing @noam-almog :-)).

**Comment by Diagoras** [comment]
I agree with both avoiding reflection and that a type safe solution would be tricky, which doesn't leave a lot to work with I guess. The Shapeless solution looks good though, if it's not too much trouble.
